op-dispute-mon/mon/resolution/delay.go

```diff
@@ -0,0 +1,53 @@
+package resolution
+
+import (
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	monTypes "github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+)
+
+type DelayMetrics interface {
+	RecordClaimResolutionDelayMax(delay float64)
+}
+
+type DelayCalculator struct {
+	metrics DelayMetrics
+	clock   clock.Clock
+}
+
+func NewDelayCalculator(metrics DelayMetrics, clock clock.Clock) *DelayCalculator {
+	return &DelayCalculator{
+		metrics: metrics,
+		clock:   clock,
+	}
+}
+
+func (d *DelayCalculator) RecordClaimResolutionDelayMax(games []monTypes.EnrichedGameData) {
+	var maxDelay uint64 = 0
+	for _, game := range games {
+		delay := d.getMaxDelay(&game)
```

Comment:
Seems weird to take an array of non-pointer `EnrichedGameData` and then have to create a pointer here.  Seems like either `getMaxDelay` should take a non-pointer type or the array should be an array of pointer types. ie let's be consistent in either passing by reference or by value.

Reply:
Very good point, I fixed up the dispute monitor to use pointers instead of passing in the `EnrichedGameData` arrays by value which should perform quite a bit better since we aren't copying the arrays for the `{delayCalculator, forecaster, detector}`.